compiler/src/main/java/org/qbicc/type/CompoundType.java

```diff
@@ -34,11 +34,9 @@ public final class CompoundType extends ValueType {
     private volatile List<Member> members;
     private volatile List<Member> paddedMembers;
     private volatile Map<String, Member> membersByName;
-    // late-computed hash code
-    private int hashCode;
 
     CompoundType(final TypeSystem typeSystem, final Tag tag, final String name, final Supplier<List<Member>> membersResolver, final long size, final int overallAlign) {
-        super(typeSystem, 0);
+        super(typeSystem, CompoundType.class.hashCode());
```

Comment:
We do not want the hash code of every `CompoundType` to be the same for performance reasons (we key maps by type with some regularity). How about leaving this as 0 but overriding `hashCode` to return `System.identityHashCode(this)`?

Reply:
I was worried about breaking the invariant that objects that are `equal` should have the same `hashCode`.

Hashing on `name` doesn't work, because sometimes that is allowed to be `null`.  Maybe name (when non-null), size, number of members, and align would be enough?

Reply:
Changed to include name, size, align in hashcode.